src/main/java/org/opensearch/neuralsearch/highlight/SemanticHighlighter.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.neuralsearch.highlight;
+
+import lombok.extern.log4j.Log4j2;
+import org.opensearch.index.mapper.MappedFieldType;
+import org.opensearch.neuralsearch.ml.MLCommonsClientAccessor;
+import org.opensearch.search.fetch.subphase.highlight.FieldHighlightContext;
+import org.opensearch.search.fetch.subphase.highlight.HighlightField;
+import org.opensearch.search.fetch.subphase.highlight.Highlighter;
+import org.opensearch.core.common.text.Text;
+import org.opensearch.neuralsearch.highlight.extractor.QueryTextExtractorRegistry;
+
+/**
+ * Semantic highlighter that uses ML models to identify relevant text spans for highlighting
+ */
+@Log4j2
+public class SemanticHighlighter implements Highlighter {
+    public static final String NAME = "semantic";
+
+    private SemanticHighlighterEngine semanticHighlighterEngine;
+
+    public void initialize(MLCommonsClientAccessor mlClient) {
```

Comment:
Better to receive `SemanticHighlighterEngine` directly for easier unit test.
```suggestion
    public void initialize(SemanticHighlighterEngine semanticHighlighterEngine) {
```

Reply:
Updated.